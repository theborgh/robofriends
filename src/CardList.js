import React from 'react';
import Card from './Card';

const CardList = ({robotList}) => {

    return (
        <div>
            {
            robotList.map(rob => 
                <Card key={rob.id} id={rob.id} name={rob.name} username={rob.username} email={rob.email} />)
            }
        </div>
    );
}

export default CardList;