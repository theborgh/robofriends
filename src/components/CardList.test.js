import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it("renders cardlist component", () => {
  const mockRobots = [
    {
      id: 1,
      name: "Jimmy",
      email: "jimmy@gmail.com"
    }
  ]
  expect(shallow(<CardList robotList={mockRobots} />)).toMatchSnapshot();
})