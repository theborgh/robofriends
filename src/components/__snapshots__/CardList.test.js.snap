// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cardlist component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robotList={
      Array [
        Object {
          "email": "jimmy@gmail.com",
          "id": 1,
          "name": "Jimmy",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="jimmy@gmail.com"
          id={1}
          name="Jimmy"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "jimmy@gmail.com",
          "id": 1,
          "name": "Jimmy",
          "username": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="jimmy@gmail.com"
            id={1}
            name="Jimmy"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "jimmy@gmail.com",
            "id": 1,
            "name": "Jimmy",
            "username": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
